projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -807,21 +812,24 @@ export class IgxOverlayService implements OnDestroy {
         }
     }
 
-    private removeOutsideClickListener(info: OverlayInfo) {
-        if (info.settings.modal === false) {
```

Comment:
when the overlay is `modal` we do not add event to the `document` but to the modal overlay element. Do not remove this check!
Instead, we should remove the handler from the overlay element. Look at the `addOutsideClickListener` to see where the event is added when the overlay is `modal`